### ISSUE TITLE
Time picker in igx-datePicker

### ISSUE THREAD

**Issue by Eralmidia** [IgniteUI/igniteui-angular#520]
### Description  
Support for time in igx-datePicker.

The date picker is hardly usuable for out team without the ability to select time as well.

**Comment by StefanIvanov** [comment]
Hello Eralmidia, what kind of time picking mechanisms does your team use within the usual date pickers they are used to? Also do you use a date picker in a date only or time only mode or you have both modes combined within the same instance of the control? Screenshots will be appreciated!

**Comment by Eralmidia** [comment]
I have to say, I really like how responsive you guys are. Yesterday you fixed a bug within a day, and now you respond to feature requests within 10 minutes. Thumbs up.

I guess there are several ways to go. The easiest would probably be something like PrimeNG has. If you look at the input field labeled Time on this page: https://www.primefaces.org/primeng/#/calendar. The only required input would probably be a choice between 12/24 hour format.

Then of course, you have a bit more elegant (and probably efficient) solutions, like "android style" time picking like this https://android-arsenal.com/details/1/123.

When it comes to modes, our need is most commonly that the user can pick an exact point in time (so a date and time picking togehter in the same input field). But of course, I'm sure the value of igx-datwPicker would be even greater if you could choose between the three modes.

**Comment by StefanIvanov** [comment]
Thank you for the input. Your feature request is already on our plate and the feedback above is really important as it will drive not only the component development, but also our efforts in making sure that it meets the requirements and expectations of our users. Thumbs up for your proactivity!

**Comment by Eralmidia** [comment]
Good to hear. I couldn't see this in the road map doc, so I'm guessing it's not Q1?

**Comment by kdinev** [comment]
@Eralmidia We need to evaluate it further in order to update the roadmap. This will most probably be happening next week. We will keep you posted on the progress.

**Comment by StefanIvanov** [comment]
Draft of the Time Picker spec is available [here](https://github.com/IgniteUI/igniteui-angular/wiki/Time-Picker)

**Comment by Eralmidia** [comment]
Draft looks good, but I have one question: This seems (at least when you look at the sample UI) to implement the time picker as its own separate component. At least for us, the most common scenario is that the user should be able to pick a timestamp, i.e both date and time into a single text field. So for this to be useful to us, we need the ability to show both date and time picker in the same popup.

**Comment by kdinev** [comment]
@Eralmidia We decided that, given the specifics of the JavaScript `Date` object, where date is always created in the local browser time zone, it's better to separate the UI into three - picking a date, picking time and picking a time zone info, since this gives better control to the developer over how they want the date to be sent to their server. 

The approach doesn't require changes to the datepicker UI. We've tackled this problem a lot with Ignite UI and we never got to a satisfactory universal solution with the `igDatePicker`. We even had to rewrite `Date` handling last year. Reference: https://github.com/IgniteUI/ignite-ui/pull/540 

The approach would be the same as on Facebook (events):

![image](https://user-images.githubusercontent.com/1472513/35680186-46005ec6-0762-11e8-9821-aa782f574876.png)
